app/src/main/java/mn/devfest/speakers/SpeakerListFragment.java

```diff
@@ -68,24 +69,31 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         getActivity().setTitle(getResources().getString(R.string.speaker_title));
 
         mAdapter = new SpeakerListAdapter();
-        mAdapter.setSpeakers(mSpeakerData);
+        mAdapter.setSpeakers(mSpeakerList);
         mSpeakerRecyclerView.setAdapter(mAdapter);
         LinearLayoutManager mLinearLayoutManager = new LinearLayoutManager(getActivity());
         mSpeakerRecyclerView.setLayoutManager(mLinearLayoutManager);
-        //mSpeakerRecyclerView.addItemDecoration(new DividerItemDecoration(getContext(), DividerItemDecoration.VERTICAL_LIST));
-    }
-
-    @Override
-    public void onResume() {
-        super.onResume();
-        //Refresh the UI with the latest data
-        List<Speaker> speakers = mDataSource.getSpeakers();
-
-        if (speakers.size() == 0) {
-            mLoadingView.setVisibility(View.VISIBLE);
-        } else {
-            setSpeakers(speakers);
-        }
+        mSpeakerRecyclerView.addItemDecoration(new DividerItemDecoration(getContext(), DividerItemDecoration.VERTICAL_LIST));
+        mFirebaseDatabaseReference = FirebaseDatabase.getInstance().getReference();
+        mFirebaseDatabaseReference.child(DEVFEST_2017_KEY).child(SPEAKERS_CHILD_KEY)
```

Comment:
I've seen this block a few times- have you considered moving it into something like a manager/service class to eliminate the duplication?